bitnami/redis/templates/scripts-configmap.yaml

```diff
@@ -491,6 +491,10 @@ data:
     [[ -f "$REDIS_PASSWORD_FILE" ]] && export REDISCLI_AUTH="$(< "${REDIS_PASSWORD_FILE}")"
 
     if ! failover_finished; then
+        echo "I am the master pod and you are stopping me. Stopping client connections"
+        # Stopping client write connections to avoid data loss
+        run_redis_command CLIENT PAUSE "{{ mul (sub .Values.sentinel.terminationGracePeriodSeconds 10) 1000 }}" WRITE
+        # if I am the master, issue a command to failover once and then wait for the failover to finish
         echo "Waiting for sentinel to run failover for up to {{ sub .Values.sentinel.terminationGracePeriodSeconds 10 }}s"
         retry_while "failover_finished" "{{ sub .Values.sentinel.terminationGracePeriodSeconds 10 }}" 1
```

Comment:
While issuing a "client pause" prevents write losses until we notice the role switch here, this can/will take longer than necessary:

The "SENTINEL FAILOVER" command just assumes that the master is dead. This means that it does not actively reconfigure the master as part of the failover. It will promote one of the replicas to become a master and will then reconfigure the remaining replicas of the _former_ master. 

If sentinel is asked for the master address, it will return the new master as soon as it has verified that it switched its role to master. 

However, the former master (which we query in `failover_finished`) will change its role only by the regular checks sentinel makes. Sentinel will notice that this node is supposed to be a replica and reconfigure it (after waiting some additional time).

That's why the write loss window reported in  https://github.com/bitnami/charts/issues/12598 is so big (approx. 10 seconds).

To improve this, we could use the check used by the sentinel script to find out if the failover finished (that one asks the sentinel for the master). When we shut down the former master then, reconnecting clients will connect to the new master, which is what we want.

All in all however, I wonder if it is worth the hassle. Before #6080 we did not do it anything here and, thus, shut down the master as quickly as possible, creating a time window during which clients could not reach the master.  Now we introduce a time window during which only reads are allowed, but which is longer than necessary because of the polling. Which one is better probably depends on the application.